### DISCUSSION TITLE
Is graceful shutdown working as expected?

### DISCUSSION THREAD

**Discussion by emanuel-gjini** [ntex-rs/ntex#379] [Q&A] [answered]
I was trying out Ntex, and by far it seems the best framework for what I need to do. But while testing out the graceful shutdown, it looks like it is not working as expected. It kills the connection even if it is still in-progress.

I've attached a zip with a minimal example to reproduce.
[Archive.zip](https://github.com/user-attachments/files/16165401/Archive.zip)

When you start the app in the docker container and make a request to the handler that is supposed to take 10 seconds to respond and you call `docker kill --signal SIGTERM container_name` it starts to gracefully shutdown but it kills the connection instead of waiting for it to finish.

I'have already tried to adjust the timeouts but no luck.

**Comment by fafhrd91** [comment]
at the moment web part does not support graceful shutdown, i am planing to work on this feature in couple weeks

**Reply by emanuel-gjini** [reply]
@fafhrd91 awesome, thanks for letting me know 🙌🏻

**Reply by fafhrd91** [reply, marked as answer]
ntex v2.2.0 has proper graceful shutdown feature
